src/index.js

```diff
@@ -314,11 +330,22 @@ module.exports = function ChannelsMiddleware(mwOpts) {
 
 								const ctx = Context.create(broker, null, msg, {
 									parentCtx,
-									caller,
+									caller: headers?.$parentServiceName
+										? headers.$parentServiceName
+										: caller,
 									meta,
 									headers: ctxHeaders
 								});
 
+								// attach channelName to context so in the handler we can use it
+								ctx.currentChannelName = chan.name;
+								ctx.currentServiceName = svc.fullName;
```

Comment:
It's not necessary, because it's available via `ctx.service.fullName`

Reply:
You're right. This was resolved in https://github.com/moleculerjs/moleculer-channels/pull/92/files#r2446701561